examples/rust-qfunctions/ex1-volume.c

```diff
@@ -0,0 +1,438 @@
+  CeedInt     sol_degree  = 4;               // polynomial degree for the solution
+  CeedInt     num_qpts    = sol_degree + 2;  // number of 1D quadrature points
+  CeedInt     prob_size   = -1;              // approximate problem size
+  CeedInt     help = 0, test = 0, gallery = 0, benchmark = 0;
+
+  // Process command line arguments.
+  for (int ia = 1; ia < argc; ia++) {
+    // LCOV_EXCL_START
+    int next_arg = ((ia + 1) < argc), parse_error = 0;
+    if (!strcmp(argv[ia], "-h")) {
+      help = 1;
+    } else if (!strcmp(argv[ia], "-c") || !strcmp(argv[ia], "-ceed")) {
+      parse_error = next_arg ? ceed_spec = argv[++ia], 0 : 1;
+    } else if (!strcmp(argv[ia], "-d")) {
+      parse_error = next_arg ? dim = atoi(argv[++ia]), 0 : 1;
+      num_comp_x                   = dim;
+    } else if (!strcmp(argv[ia], "-m")) {
+      parse_error = next_arg ? mesh_degree = atoi(argv[++ia]), 0 : 1;
+    } else if (!strcmp(argv[ia], "-p")) {
+      parse_error = next_arg ? sol_degree = atoi(argv[++ia]), 0 : 1;
+    } else if (!strcmp(argv[ia], "-q")) {
+      parse_error = next_arg ? num_qpts = atoi(argv[++ia]), 0 : 1;
+    } else if (!strcmp(argv[ia], "-s")) {
+      parse_error = next_arg ? prob_size = atoi(argv[++ia]), 0 : 1;
+    } else if (!strcmp(argv[ia], "-b")) {
+      parse_error = next_arg ? benchmark = atoi(argv[++ia]), 0 : 1;
+    } else if (!strcmp(argv[ia], "-t")) {
+      test = 1;
+    } else if (!strcmp(argv[ia], "-g")) {
+      gallery = 1;
+    }
+    if (parse_error) {
+      printf("Error parsing command line options.\n");
+      return 1;
+    }
+    // LCOV_EXCL_STOP
+  }
+  if (prob_size < 0) prob_size = test ? 8 * 16 : 256 * 1024;
+
+  // Print the values of all options:
+  if (!test || help) {
+    // LCOV_EXCL_START
+    printf("Selected options: [command line option] : <current value>\n");
+    printf("  Ceed specification     [-c] : %s\n", ceed_spec);
+    printf("  Mesh dimension         [-d] : %" CeedInt_FMT "\n", dim);
+    printf("  Mesh degree            [-m] : %" CeedInt_FMT "\n", mesh_degree);
+    printf("  Solution degree        [-p] : %" CeedInt_FMT "\n", sol_degree);
+    printf("  Num. 1D quadrature pts [-q] : %" CeedInt_FMT "\n", num_qpts);
+    printf("  Approx. # unknowns     [-s] : %" CeedInt_FMT "\n", prob_size);
+    printf("  QFunction source       [-g] : %s\n", gallery ? "gallery" : "header");
+    if (help) {
+      printf("Test/quiet mode is %s\n", (test ? "ON" : "OFF (use -t to enable)"));
+      return 0;
+    }
+    printf("\n");
+    // LCOV_EXCL_STOP
+  }
+
+  // Select appropriate backend and logical device based on the (-ceed) command line argument.
+  Ceed ceed;
+
+  CeedInit(ceed_spec, &ceed);
+
+  // Add the path to the Rust crate to the ceed object.
+    {
+      char  root[2048] = __FILE__;
+      char *last_slash = strrchr(root, '/');
+      strncpy(last_slash + 1, "ex1-volume-rs", 14);
+      CeedAddRustSourceRoot(ceed, root);
```

Comment:
```suggestion
      char  root[2048] = __FILE__;
      char *last_slash = strrchr(root, '/');
 
      strncpy(last_slash + 1, "ex1-volume-rs", 14);
      CeedAddRustSourceRoot(ceed, root);
```
minor - blank line after variable declaration block